qt/HelloQt.hpp

```diff
@@ -0,0 +1,12 @@
+#include <QWidget>
+
+class HelloQt : public QWidget
+{
+	Q_OBJECT // Very Important, this is all the magic that is needed
+	public:
+		explicit HelloQt( QWidget* parent = nullptr );
+		~HelloQt();
+	private slots:
+		void writeToDebug( void );
+
```

Comment:
No need to specify explicit on line 7 or void on line 10

Reply:
Sorry Paul, in retrospect I think you're right about explicit usage here. I've just been following the style in the Qt examples, but after a bit of googling the general (non Qt-specific) advice is to use explicit with constructors with one argument.